sklearn/gaussian_process/gpr.py

```diff
@@ -241,9 +242,15 @@ def obj_func(theta, eval_gradient=True):
         # of actual query points
         K = self.kernel_(self.X_train_)
         K[np.diag_indices_from(K)] += self.alpha
-        self.L_ = cholesky(K, lower=True)  # Line 2
+        try:
+            self.L_ = cholesky(K, lower=True)  # Line 2
+        except np.linalg.LinAlgError:
+            raise ValueError("The kernel, %s, isn't returning a "
```

Comment:
In order to keep the original traceback, it is better to modify the exception and raise the original exception. I'll do it and push into your branch.